### PR TITLE
spacing class mixin and generation

### PR DESCRIPTION
**Pull request by jjbert18**
up and working with full demo on codepen http://codepen.io/CodeAnd_/full/JEwVmq/
any idea on how to minify this or the demo would be great

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,6 +111,10 @@ This breakpoint mixin gives you the flexibility to use any breakpoints you want
 
 Cross browser support (`-ms-, -webkit-, and -moz`) support for styling the input placeholder.
 
+#### [Spacing Classes](css/sass/spacing-classes.scss)
+
+A mixin to generate easy-to-use margin and padding classes for all and any combination of sides in multiple units (rem, em, and px). Demo can be found on CodePen @ http://codepen.io/CodeAnd_/full/JEwVmq/
```
Comment (andrewicarlson) [inline README.md]: Please wrap "CodePen" in the link instead of pasting it raw.

File: css/sass/spacing-classes.scss [hunk 2]
```diff
@@ -0,0 +1,100 @@
+/**
+ * @margin-and-padding.scss
+ * global classes for margin and padding
+ * naming order follows the property shorthand:
+ * • top • right • bottom • left
+ * m or p (margin or padding)
+ * side  (top • right • bottom • left)
+ * amount and unit (px • em • rem)
+ * example generated classes:
+ *   .m-trbl--1px, .m-t--1em, m-b--1rem
```
Comment (andrewicarlson) [inline css/sass/spacing-classes.scss:10]: I think class names like `.m-trbl--1px` are rather obtuse. 

What is the benefit of generating all of those extra classes when you could just do: `.m-t--1px.m-b--1px` etc? That also gives you more flexibility to change it up like `.m-t--1em.m-b--2px`

**Review by andrewicarlson — CHANGES_REQUESTED** [review]
I think chaining classes may be better than long obtuse class names that force the same sizes. Thoughts?

File: css/sass/spacing-classes.scss (hunk 2 above)
Comment (xerik25) [reply css/sass/spacing-classes.scss:10]: That makes sense. I think we were just trying to get fancy with the generator.

Comment (jjbert18) [reply css/sass/spacing-classes.scss:10]: The benefit is adding less classes to the html. While you can certainly chain it if you so choose, I think having options for all groups seems easier.

`.class.m-t--1rem.m-r--1rem.m-b--1rem.m-l--1rem` vs `.class.m-trbl-1rem` 

Can and be used in a combo `.class.m-trb--5rem.m-l--10rem`

Comment (andrewicarlson) [reply css/sass/spacing-classes.scss:10]: I disagree with that specific implementation but I guess it's fine to leave it in. It's up to whoever uses it to strip out what they don't need. 

If you could just fix the link in the README.md we'll get this merged!

**Review by andrewicarlson — APPROVED** [review]

**Pull request merged**